mantle/kola/tests/misc/network.go

```diff
@@ -293,35 +363,141 @@ var (
+		"ignition": {
+			"version": "3.2.0"
+		},
+		"storage": {
+			"files": [
+				{
+					"path": "/etc/systemd/network/99-default.link",
+					"contents": { "source": "data:text/plain;base64,%s" },
+					"mode": 420
+				},
+				{
+					"path": "/etc/NetworkManager/conf.d/10-dhcp-config.conf",
+					"contents": { "source": "data:text/plain;base64,%s" },
+					"mode": 420
+				},
+				{
+					"path": "/usr/local/bin/capture-macs",
+					"contents": { "source": "data:text/plain;base64,%s" },
+					"mode": 755
+				},
+				{
+					"path": "/usr/local/bin/create-veth-pairs",
+					"contents": { "source": "data:text/plain;base64,%s" },
+					"mode": 755
+				},
+				{
+					"path": "/usr/local/bin/setup-ovs",
+					"contents": { "source": "data:text/plain;base64,%s" },
+					"mode": 755
+				}
+			]
+		},
+		"systemd": {
+			"units": [
+				{
+					"enabled": true,
+					"name": "openvswitch.service"
+				},
+				{
+					"contents": "[Unit]\nDescription=Capture MAC address from kargs\nAfter=network-online.target\nAfter=openvswitch.service\nConditionKernelCommandLine=macAddressList\n\n[Service]\nType=oneshot\nExecStart=/usr/local/bin/capture-macs\n\n[Install]\nRequiredBy=multi-user.target\n",
+					"enabled": true,
+					"name": "capture-macs.service"
+				},
+				{
+					"contents": "[Unit]\nDescription=Create VETH Pairs and Configue OVS interface over bond\nAfter=network-online.target\nAfter=capture-macs.service\nAfter=openvswitch.service\n\n[Service]\nType=oneshot\nExecStart=/usr/local/bin/create-veth-pairs\n\n[Install]\nRequiredBy=multi-user.target\n",
+					"enabled": true,
+					"name": "create-veth-pairs.service"
+				}
+			]
+		}
+	}`,
+		base64.StdEncoding.EncodeToString([]byte(defaultLinkConfig)),
+		base64.StdEncoding.EncodeToString([]byte(dhcpClientConfig)),
+		base64.StdEncoding.EncodeToString([]byte(captureMacsScript)),
+		base64.StdEncoding.EncodeToString([]byte(setupVethPairs)),
+		base64.StdEncoding.EncodeToString([]byte(setupOvsScript))))
+
+	switch pc := c.Cluster.(type) {
+	// These cases have to be separated because when put together to the same case statement
+	// the golang compiler no longer checks that the individual types in the case have the
+	// NewMachineWithQemuOptions function, but rather whether platform.Cluster
+	// does which fails
+	case *unprivqemu.Cluster:
+		m, err = pc.NewMachineWithQemuOptions(userdata, options)
+	default:
+		panic("unreachable")
+	}
+	if err != nil {
+		c.Fatal(err)
+	}
+
+	// Add karg needed for the ignition to configure the network properly.
+	addKernelArgs(c, m, []string{fmt.Sprintf("macAddressList=%s,%s", primaryMac, secondaryMac)})
+}
+
 // NetworkSecondaryNics verifies that secondary NICs are created on the node
 func NetworkSecondaryNics(c cluster.TestCluster) {
 	primaryMac := "52:55:00:d1:56:00"
 	secondaryMac := "52:55:00:d1:56:01"
 
 	setupMultipleNetworkTest(c, primaryMac, secondaryMac)
 
-	checkOvsBridge(c, primaryMac)
+	checkOvsBridge(c, primaryMac, "")
```

Comment:
Could you create a new function for this instead? One that would set the primary IP, one that would not care about it. They can use the same helper function inside. Having dedicated function names would make it more clear and readable.

Reply:
changed the function to `checkOvsBridgeParams`, then created 2 wrapper functions `checkBondReceivedExpectedMAC` `checkBondReceivedExpectedMACAndIP`. 
Is it readable enough?